packages/flutter_tools/test/integration.shard/migrate_utils_test.dart

```diff
@@ -0,0 +1,223 @@
+
+void main() {
+  BufferLogger logger;
+  FileSystem fileSystem;
+  Directory projectRoot;
+  String projectRootPath;
+  MigrateUtils utils;
+  ProcessUtils processUtils;
+
+  setUpAll(() async {
+    fileSystem = globals.localFileSystem;
+    logger = BufferLogger.test();
+    utils = MigrateUtils(
+      logger: logger,
+      fileSystem: fileSystem,
+      platform: globals.platform,
+      processManager: globals.processManager,
+    );
+    processUtils = ProcessUtils(processManager: globals.processManager, logger: logger);
+  });
+
+  group('git', () {
+    setUp(() async {
+      projectRoot = fileSystem.systemTempDirectory.createTempSync('flutter_migrate_utils_test');
+      projectRoot.createSync(recursive: true);
+      projectRootPath = projectRoot.path;
+    });
+
+    testWithoutContext('init', () async {
+      expect(projectRoot.existsSync(), true);
+      expect(projectRoot.childDirectory('.git').existsSync(), false);
+      await utils.gitInit(projectRootPath);
+      expect(projectRoot.childDirectory('.git').existsSync(), true);
+    });
+
+    testWithoutContext('isGitIgnored', () async {
+      expect(projectRoot.existsSync(), true);
+      expect(projectRoot.childDirectory('.git').existsSync(), false);
+      await utils.gitInit(projectRootPath);
+      expect(projectRoot.childDirectory('.git').existsSync(), true);
+
+      projectRoot.childFile('.gitignore')
+        ..createSync()
+        ..writeAsStringSync('ignored_file.dart', flush: true);
+
+      expect(await utils.isGitIgnored('ignored_file.dart', projectRootPath), true);
+      expect(await utils.isGitIgnored('other_file.dart', projectRootPath), false);
+    });
+
+    testWithoutContext('isGitRepo', () async {
+      expect(projectRoot.existsSync(), true);
+      expect(projectRoot.childDirectory('.git').existsSync(), false);
+      expect(await utils.isGitRepo(projectRootPath), false);
+      await utils.gitInit(projectRootPath);
+      expect(projectRoot.childDirectory('.git').existsSync(), true);
+
+      expect(await utils.isGitRepo(projectRootPath), true);
+
+      expect(await utils.isGitRepo(projectRoot.parent.path), false);
+    });
+
+    testWithoutContext('hasUncommitedChanges false on clean repo', () async {
+      expect(projectRoot.existsSync(), true);
+      expect(projectRoot.childDirectory('.git').existsSync(), false);
+      await utils.gitInit(projectRootPath);
+      expect(projectRoot.childDirectory('.git').existsSync(), true);
+
+      projectRoot.childFile('.gitignore')
+        ..createSync()
+        ..writeAsStringSync('ignored_file.dart', flush: true);
+
+      await processUtils.run(<String>['git', 'add', '.'], workingDirectory: projectRootPath);
+      await processUtils.run(<String>['git', 'commit', '-m', 'Initial commit'], workingDirectory: projectRootPath);
+
+      expect(await utils.hasUncommitedChanges(projectRootPath), false);
+    });
+
+    testWithoutContext('hasUncommitedChanges true on dirty repo', () async {
```

Comment:
When I run this test locally, it passes, but logs out `fatal: bad revision 'HEAD'`

Reply:
Ahh, I still have the logging in this particular test because im still ironing out a few kinks with the command formatting as it's a bit sensitive when specifying the exclude path. This should be fixed in a bit.